rust/extractor/src/main.rs

```diff
@@ -118,3 +125,19 @@ fn main() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+fn main() -> anyhow::Result<()> {
+    let cfg = config::Config::extract().context("failed to load configuration")?;
+    let qltest = cfg.qltest;
+    let qltest_log = cfg.log_dir.join("qltest.log");
+    let result = std::panic::catch_unwind(|| run_extractor(cfg));
+    if qltest && matches!(result, Err(_) | Ok(Err(_))) {
+        // in case of failure, print out the full log
+        let log = File::open(qltest_log).context("opening qltest.log")?;
+        let reader = BufReader::new(log);
+        for line in reader.lines() {
+            println!("{}", line.context("reading qltest.log")?);
+        }
+    }
+    result.unwrap()
```

Comment:
Unwrap seems bad here. If there was a caught `panic` then it would be ignored and replaced by an uninformative panic due to unwrapping an `Err`